src/robot_upstart/install_script.py

```diff
@@ -91,6 +94,10 @@ def main():
     if args.augment:
         j.generate_system_files = False
 
-    j.install()
+    provider=providers.Upstart
+    if args.systemd:
+        provider=providers.Systemd
+
+    j.install(Provider=provider)
```

Comment:
Ideally we would detect the init system here and just do the right thing, but having looked into it previously, I acknowledge that detecting it reliably can be tricky. Other possibility for an interface would be `--provider systemd`, but I'm not super attached to that; the flag is fine.